kong/clustering/wrpc_control_plane.lua

```diff
@@ -60,6 +61,7 @@ end
 local function get_config_service(self)
   if not wrpc_config_service then
     wrpc_config_service = wrpc.new_service()
+    version_negotiation.add_negotiation_service(wrpc_config_service, self.conf)
     wrpc_config_service:add("kong.services.config.v1.config")
```

Comment:
What prevents a CP node to execute a ConfigService.SyncConfig before the negotiation is finished?

Reply:
The same question for the DP.

Reply:
nothing yet.  i could put a "negotiated" flag in the peer object and just drop everything else until then.